lib/hammer_cli_foreman_admin/logging_command.rb

```diff
@@ -0,0 +1,127 @@
+    rescue Exception => e
+      if raise_on_error
+        raise "Command '#{cmd}' failed: #{e}"
+      else
+        print_message "Command '#{cmd}' failed: #{e}"
+      end
+    end
+
+    def check_options(hash, action_name, *required_keys)
+      required_keys.each do |key|
+        raise("Missing option '#{key}' in action '#{action_name}' component '#{hash[:name]}'") unless hash[key]
+      end
+    end
+
+    def action_functions
+      @action_functions ||= {
+        :run_command => lambda do |opts|
+          check_options opts, :run_command, :command
+          run_command opts[:command], false
+        end,
+        :run_command_on_file => lambda do |opts|
+          check_options opts, :run_command_on_file, :command, :file
+          run_command "#{opts[:command]} #{opts[:file]}", false
+        end,
+        :create_file => lambda do |opts|
+          check_options opts, :create_file, :file, :contents
+          FileUtils.mkdir_p(File.dirname(opts[:file]))
+          open(opts[:file], 'w') { |f| f.puts opts[:contents] }
+        end,
+        :remove_file => lambda do |opts|
+          check_options opts, :remove_file, :file
+          File.unlink(opts[:file])
+        end,
+        :ensure_line_is_present => lambda do |opts|
+          check_options opts, :ensure_line_is_present, :file, :line
+          if File.foreach(opts[:file]).grep(/#{opts[:line][0]}/).empty?
+            open(opts[:file], 'a') { |f| f.puts "\n" + opts[:line].join }
+          else
+            run_command %Q|sed -i 's!#*#{opts[:line][0]}\s*#{opts[:line][1]}.*!#{opts[:line].join}!' #{opts[:file]}|
+          end
+        end
+      }
+    end
+
+    def new_level
+      option_level_debug? ? :debug : :production
+    end
+
+    def configure_component(component, level)
+      name = component[:name]
+      friendly_name = component[:friendly_name]
+      file = component[:file]
+      backup_suffix = Time.now.utc.to_i.to_s(36)
+      if File.exists?(file)
+        component[level].each do |action|
+          action_name = action[:action]
+          action[:name] = name
+          action[:file] = file unless action[:file]
+          func = action_functions[action_name.to_sym]
+          if func
+            logger.info "Processing #{name} action #{action_name}"
+            unless option_dry_run?
+              unless option_no_backup?
+                backup_file = "#{file}.#{backup_suffix}~"
+                logger.info "Creating backup #{backup_file}"
+                FileUtils.cp(file, backup_file)
+              end
+              func.call(action)
+            end
+          else
+            raise "Unknown action #{action_name} for component #{name}"
+          end
+        end
+      else
+        logger.info "Skipped component #{name}, file #{file} does not exist"
+      end
+    end
+
+    def list_components
+      HammerCLI::Settings.get(:admin)[:logging][:component].each do |component|
+        print_message(component[:name])
+      end
```

Comment:
Have you considered printing the values in a table? I don't care about formatting but it would enable for printing the list in csv and json (`--output=json`).

The code would be something like:
```ruby
output_definition = HammerCLI::Output::Definition.new
output_definition.fields << Fields::Field.new(:label => _('Component'), :path => ["name"])
output = HammerCLI::Output::Output.new(context, :default_adapter => :table)
output.print_collection(output_definition, HammerCLI::Settings.get(:admin)[:logging][:component])
```